compiler/tests-spec/testData/diagnostics/notLinked/dfa/neg/24.fir.kt

```diff
@@ -68,8 +68,8 @@ fun case_4(x: String?) {
             break
         }
     }
-    <!DEBUG_INFO_EXPRESSION_TYPE("kotlin.String & kotlin.String?")!>x<!>
-    <!DEBUG_INFO_EXPRESSION_TYPE("kotlin.String & kotlin.String?")!>x<!>.length
+    <!DEBUG_INFO_EXPRESSION_TYPE("kotlin.String?")!>x<!>
+    <!DEBUG_INFO_EXPRESSION_TYPE("kotlin.String?")!>x<!>.<!INAPPLICABLE_CANDIDATE!>length<!>
```

Comment:
Ah, but this and below ones look like good refinements, i.e., can be part of 4th commit.